### ISSUE TITLE
Incoming calls ring through earpiece when phone is in locked state

### ISSUE THREAD

**Issue by soloclub** [WhisperSystems/Signal-Android#6356]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------
**Bug description**
Incoming signal calls ring through earpiece when phone is in locked state. When tried with called party phone unlocked, rings through speaker without any issue.

#**Steps to reproduce**
- Lock phone "A" and call from phone "B"
OR
- Lock phone "B" and call from phone "A"

**Actual result**
Locked phone rings through earpiece.

**Expected result:** 
Locked device should ring through speaker.

Device info

Device "A" 
Nexus 5
Android version: 7.1.1 AOSP  (No Gapps)
Signal version: 3.31.0

Device "B" 
YU Yureka
Android version: cm 13.0 (No Gapps)
Signal version: 3.31.0

**Comment by chrihu** [comment]
Lets add some information to the issue so it hopefully can get solved, as it makes the whole calling part of signal a bit unfeasible.
Testdevices: 2x Nexus 5 (aosp 7.1.1 and stock 6.0.1) and HTC Desire S (4.0.4) Signal 3.31.4

If there is no lockscreen enabled the calls ring through the speaker as they should even when the screen is off, but as soon as there is any lockscreen set, calls ring trough the earpiece.
It always rings through the earpiece when a lockscreen is enabled and the screen is off, but there is a difference between 6.0.1 and 7.1.1 when the screen is on.
On 6.0.1 it rings trough the earpiece when the display is on and a lockscreen is enabled, on 7.1.1 it rings through the speaker as it should in this scenario.

logcats on demand

**Comment by mykhi** [comment]
Same issue here
Buggy device: Samsung S7 edge (SM0G93F) rooted @Android 7.0, Signal: 4.0.0

**Comment by xenithorb** [comment]
Same: Bacon, Android 7.1.1, Signal 4.2.4

**Comment by janvlug** [comment]
This phone has also this issue: bq Aquaris 35 (bg_Aquaris3), Android 4.2.2, Signal 4.2.4
Messaging and calling from an OnePlus X to the bq Aquaris results on the bq Aquaris:
When screen is locked and off: both incoming message sound notification and call ringing sound are through ear-speaker.
When screen is unlocked and on: incoming message sound is through ear-speaker. Incoming ring sound is 1/2 second through normal/hands-free speaker and continues after that through ear-speaker

**Comment by janvlug** [comment]
Today I noticed that the notification sound on my bq Aquaris 35 was working again when receiving messages. I did some tests and could not reproduce the sound through the earpiece any more. The ringing sound when calling still is through the earpiece (only the first 1/2 second normal). I'm currently using Signal 4.2.5.

**Comment by janvlug** [comment]
I was testing a bit further when sending messages from the Signal desktop to myself. Sometimes I do not get the notification sound, sometimes I do get it. I could not reliable reproduce the behavior, however it seems that switching sound profiles can trigger the issue that I do not get the notification sounds anymore. My phone has four profiles. changing between 'Outdoor' and 'General' seems sometimes to influence whether there is a notification sound. Maybe the condition whether the phone screen is on of off as well. I also noticed that the notification sound sometimes appears only after ~ 5 seconds after that the message is already visible on the screen. I was also switching between conversations.
Please let me know if you have any further test suggestions.
Here is a log file (with log messages about delayed notifications):
https://gist.github.com/anonymous/8b9beb8f52f678dac4b1ad10bbec916d
I'm not really sure whether I should have created a separate issue for this. If so, let me know and I will do it.

**Comment by hanswundersam** [comment]
Can confirm the problem on Nexus 5 with LineageOS 14.1-2070328 (Android 7.1.1) rooted, no Gapps using Signal 4.2.5.

**Comment by scienmind** [comment]
Same behavior (quiet while locked) on:
   Nexus 6P, stock Android 7.1.2, Signal 4.2.5
   OnePlus One, CM-13 (Android 6), Signal 4.2.5
   OnePlus 3T, stock Android 7.1.1, Signal 4.2.4

Note: on Nexus 6P the ringing starts "appropriately loud" for a split second from stereo speakers, then continues quietly from the earpiece.
And as mentioned above, if the device is unlocked the ringing is loud and clear.

(That's really unfortunate, because since the WebRTC release, the audio quality is way better than the "normal" cell phone calls, so I try to use it on a daily basis, but unfortunately this is quite a blocking issue... noone answers... ¯\_(ツ)_/¯ )

**Comment by riyapenn** [comment]
Log for Motorola Photon Q XT897, Android 4.1.2, Signal 4.5.3: https://gist.github.com/anonymous/0038288dd3468a1dba288d9380f509e9

**Comment by bl-kl2** [comment]
Same here, in lock state.
2 Android phones, different behaviours, both with Signal 4.6.1

BlackBerry KEYone [Nougat 7.1.1] -> it does ring from the earpiece, so, WRONG.
Sony Xperia X Compact [Nougat 7.0] -> it does ring from the hands-free speaker, so, OK.

Below 2 tests, both done accepting the call and then hanging up immediately.

Debug logs when X Compact calls KEYone [ring from the earpiece, WRONG]:
X Compact:
[https://gist.github.com/822d74fa1f0b4fdc6b3ec12785e903d9](https://gist.github.com/822d74fa1f0b4fdc6b3ec12785e903d9)
KEYone:
[https://gist.github.com/6415c159a75caa852483e531515be2c9](https://gist.github.com/6415c159a75caa852483e531515be2c9)


Debug logs when KEYone calls X Compact [ring from the hands-free speaker, OK]:
KEYone:
[https://gist.github.com/29e657f2922143500462dc41ae2430e1](https://gist.github.com/29e657f2922143500462dc41ae2430e1)
X Compact:
[https://gist.github.com/9171e24b7469446ed2fd56e571f77604](https://gist.github.com/9171e24b7469446ed2fd56e571f77604)
